src/card_operations/card_delete_share.c

```diff
@@ -163,8 +164,7 @@ card_error_type_e card_delete_share(card_delete_share_cfg_t *delete_config) {
       }
     }
 
-    if (CARD_OPERATION_CARD_REMOVED == card_data.error_type ||
-        CARD_OPERATION_RETAP_BY_USER_REQUIRED == card_data.error_type) {
+    if (CARD_OPERATION_RETAP_BY_USER_REQUIRED == card_data.error_type) {
```

Comment:
Doesn't this remove the condition for continuing when a `CARD_OPERATION_CARD_REMOVED` error occurs?

Reply:
'CARD_OPERATION_CARD_REMOVED' will never be returned from card internal

Reply:
It will be returned from `card_handle_errors` it needs to be handled in the caller's while loop. This is not required for flows which only require applet initialization but for other cases, teardown can occur in middle of data exchange.

Reply:
Reverted, please re-review